codegen/lib/graphql_java_gen/templates/APISchema.java.erb

```diff
@@ -319,12 +323,12 @@ public class <%= schema_name %> {
                     <%= generate_build_input_code(escape_reserved_word(field.camelize_name), field.type) %>
                   <% end %>
                   <% type.optional_input_fields.each do |field| %>
-                    if (this.<%= field.camelize_name %>Seen) {
+                    if (this.<%= escape_reserved_word(field.camelize_name) %>.defined || this.<%= escape_reserved_word(field.camelize_name) %>.value != null) {
```

Comment:
Does the second part of this conditional make sense? If `defined` is false but there is a value I would argue that we shouldn't serialize it

Reply:
what do you mean? if defined is false OR value not null then we serialize. There won't be case when we have defined == false and it has a value as there is no such constructor in `Input`

Reply:
Good point @wesleyjellis. The second part does not make sense anymore after we simplified the Input class. Just have quick discussion with @sav007 and we agreed 👍